src/arrayMethodJoin.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let separ = separator;
+    let str = '';
+
+    if (separ === undefined) {
+      separ = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
```

Comment:
We can do action in case when has value, no need to rewrite
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
        str+= this[i];
      }
```